src/colors/parser.rs

```diff
@@ -724,3 +724,122 @@ mod tests {
         );
     }
 }
+
+pub fn oklab(input: &str) -> IResult<&str, Color> {
+    let (input, _) = tag("oklab(")(input)?;
+
+    //lightness can either be an percentage or a number between 0 and 100
+    let (input, l) = terminated(
+        whitespace(alt((
+            map(whitespace(parse_percentage), |percentage| percentage),
+            nom::number::complete::float,
+        ))),
+        opt(whitespace(separator)),
+    )(input)?;
+
+    //both a and b can either be an percentage between -100% and 100% or a number between -0.4 and 0.4
+    let (input, ok_a_b) = many_m_n(
+        2,
+        2,
+        terminated(
+            whitespace(alt((
+                map(alt((parse_percentage, percentage)), |percentage| {
+                    percentage * 0.4
+                }),
+                nom::number::complete::float,
+            ))),
+            opt(whitespace(separator)),
+        ),
+    )(input)?;
+
+    let (input, alpha) = opt(map(
+        whitespace(alt((percentage, relative_percentage))),
+        |percent| (percent * 255f32) as u8,
+    ))(input)?;
+
+    let (input, _) = opt(whitespace(tag(")")))(input)?;
+
+    let color = Color::from_oklab(
+        l.clamp(0.0, 1.0),
+        ok_a_b[0].clamp(-0.4, 0.4),
+        ok_a_b[1].clamp(-0.4, 0.4),
+        alpha.unwrap_or(255),
+    );
+
+    Ok((input, color))
+}
+
+#[cfg(test)]
+mod parse_oklab {
+    use super::*;
+
+    #[test]
+    fn parses_oklab() {
+        assert_eq!(
+            Ok(("", Color::rgb(46, 52, 64))),
+            oklab("32%, -0.003600, -0.023222")
+        );
+    }
+}
+
+pub fn oklch(input: &str) -> IResult<&str, Color> {
+    let (input, _) = tag("oklch(")(input)?;
+
+    //lightness can either be an percentage or a number between 0 and 100
+    let (input, l) = terminated(
```

Comment:
Same as above, with using single character variable names.

Reply:
changed to full word variable names